solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.SolrInputField;
+
+import java.util.Map;
+
+/**
+ * Batch SolrSearch Sink Plugin - Writes data to a Sinlge Node Solr or to SolrCloud.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("SolrSearch")
+@Description("This plugin allows users to build the pipelines to write data to Solr. " +
+  "The input fields coming from the previous stage of the pipeline are mapped to Solr fields. User can also specify " +
+  "the mode of the Solr to connect to. For example, Single Node Solr or SolrCloud.")
+public class SolrSearchSink extends BatchSink<StructuredRecord, SolrInputField, SolrInputDocument> {
+
+  private final SolrSearchSinkConfig config;
+  private String uniqueKey;
+  private Map<String, String> outputFieldMap;
+  private SolrClient solrClient;
+
+  public SolrSearchSink(SolrSearchSinkConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext batchSinkContext) throws Exception {
+    // Setting outputformat class in order to resolve the 'Output directory not set' error.
+    Job job = batchSinkContext.getHadoopJob();
+    job.setOutputFormatClass(NullOutputFormat.class);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    SolrSearchSinkUtility.validateSolrConnectionString(config);
+    SolrSearchSinkUtility.verifySolrConfiguration(config);
+    if (inputSchema != null) {
+      SolrSearchSinkUtility.validateIdField(inputSchema, config);
+      SolrSearchSinkUtility.validateInputFieldsDataType(inputSchema);
+    }
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    uniqueKey = config.getIdField();
+    outputFieldMap = config.getOutputFieldMap();
+    solrClient = SolrSearchSinkUtility.getSolrConnection(config);
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<KeyValue<SolrInputField, SolrInputDocument>> emitter)
+    throws Exception {
+    String solrFieldName;
+    SolrInputDocument document;
+
+    SolrSearchSinkUtility.validateIdField(structuredRecord.getSchema(), config);
+    SolrSearchSinkUtility.validateInputFieldsDataType(structuredRecord.getSchema());
+
+    if (structuredRecord.get(uniqueKey) == null) {
+      return;
+    }
+    document = new SolrInputDocument();
+    for (Schema.Field field : structuredRecord.getSchema().getFields()) {
+      solrFieldName = field.getName();
+      if (outputFieldMap.containsKey(solrFieldName)) {
+        document.addField(outputFieldMap.get(solrFieldName), structuredRecord.get(solrFieldName));
+      } else {
+        document.addField(solrFieldName, structuredRecord.get(solrFieldName));
+      }
+    }
+    solrClient.add(document);
+    solrClient.commit();
```

Comment:
this type of pattern is ok for the realtime sink, but not for a batch sink.  Batch sink should be using a Hadoop output format like the SolrOutputFormat included in the solr project.